src/util/optional.h

```diff
@@ -10,49 +10,9 @@ Author: Diffblue Ltd.
 #ifndef CPROVER_UTIL_OPTIONAL_H
 #define CPROVER_UTIL_OPTIONAL_H
 
-#if defined __clang__
-  #pragma clang diagnostic push ignore "-Wall"
-  #pragma clang diagnostic push ignore "-Wpedantic"
-#elif defined __GNUC__
-  #pragma GCC diagnostic push ignore "-Wall"
-  #pragma GCC diagnostic push ignore "-Wpedantic"
-#elif defined _MSC_VER
-  #pragma warning(push)
-#endif
-#include <nonstd/optional.hpp>
-#if defined  __clang__
-  #pragma clang diagnostic pop
-  #pragma clang diagnostic pop
-#elif defined  __GNUC__
-  #pragma GCC diagnostic pop
-  #pragma GCC diagnostic pop
-#elif defined _MSC_VER
-  #pragma warning(pop)
-#endif
+#include <optional>
 
-// Swap for std::optional when switching to C++17
-template<typename T>
-using optionalt=nonstd::optional<T>; // NOLINT template typedef
-
-typedef nonstd::bad_optional_access bad_optional_accesst;
-
-using nonstd::nullopt;
-
-/// Similar to optionalt::value but in case of empty optional, generates an
-/// invariant failure instead of throwing an exception.
-template <typename T>
-T &get_value_or_abort(optionalt<T> &opt)
-{
-  PRECONDITION(opt.has_value());
-  return opt.value();
-}
-
-/// \copydoc get_value_or_abort(optionalt<T> &)
 template <typename T>
-const T &get_value_or_abort(const optionalt<T> &opt)
-{
-  PRECONDITION(opt.has_value());
-  return opt.value();
-}
+using optionalt = std::optional<T>; // NOLINT template typedef
```

Comment:
I take this is so we don't have a much more noisy PR with a lot more minute changes. But over time the PR is going to approach half `optionalt` half `std::optional`. Does it make sense to mark the module `DEPRECATED` now and do the work of porting everything to the bare `std::optional` at a later point?

Reply:
Fair point, I just don't know what the best way to mark an entire file as deprecated is. Should I just add a comment?

Reply:
You can mark the `using` statement as `[[deprecated]]`.

`[[deprecated]] typedef S* PS;` or `using PS [[deprecated]] = S*;`

```suggestion
using optionalt [[deprecated]] = std::optional<T>; // NOLINT template typedef
```

Reply:
Yeah, I think Enrico's suggestion makes sense (https://en.cppreference.com/w/cpp/language/attributes/deprecated), unless this breaks the build for some reason (the warning is promoted to an error).

Reply:
Actually it can be `using optionalt DEPRECATED(SINCE(2023, 11, 15, "use directly std::optionalt instead" = std::optional<T>; // NOLINT template typedef`

Reply:
Fixed, thank you!